src/ui/Home.js

```diff
@@ -1,68 +1,157 @@
 import React, { useEffect, useState } from "react";
 import CreateBlog from "../Components/CreateBlog";
+import CommentsModal from "../Components/CommentsModal";
+import DeleteModal from "../Components/DeleteModal";
+import EditModal from "../Components/EditModal";
 
-export default function Home(props) {
-  const itemsPerPage = 5;
-  const [currentPage, setCurrentPage] = useState(1);
-  const [totalPages, setTotalPages] = useState(1);
+export default function Home() {
   const [data, setData] = useState([]);
+  const [data2, setData2] = useState([]); // all previous blogs postes in array in lcoal storage
+  useEffect(() => {
+    const storedData = localStorage.getItem("blogs");
+    if (storedData) {
+      setData2(JSON.parse(storedData));
+    }
+  }, []);
+  const [Alldata, setAllData] = useState([]);
+  const [postid, setpostid] = useState("");
+  const [currentpost, setcurrentpost] = useState("");
+
+  const [commentModalIsOpen, setCommentModalIsOpen] = useState(false);
+  const [deleteModalIsOpen, setDeleteModalIsOpen] = useState(false);
+  const [EditModalIsOpen, setEditModalIsOpen] = useState(false);
+
+  const loggedUserID = localStorage.getItem("loggedUserID") ?? "";
+
+  const clodeEditModal = () => {
+    setEditModalIsOpen(false);
+  };
+  const clodeDeleteModal = () => {
+    setDeleteModalIsOpen(false);
+  };
+  const openCommentModal = () => {
+    setCommentModalIsOpen(true);
+  };
+
+  const closeCommentModal = () => {
+    setCommentModalIsOpen(false);
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const Alldata = JSON.parse(localStorage.getItem("blogs"));
+        setAllData(Alldata);
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
 
+    fetchData();
+  }, []);
   useEffect(() => {
     const fetchData = async () => {
       try {
         const response = await fetch(
-          `https://jsonplaceholder.typicode.com/posts?_page=${currentPage}&_limit=${itemsPerPage}`
+          `https://jsonplaceholder.typicode.com/posts`
         );
         const jsonData = await response.json();
         setData(jsonData);
-        const totalItems = parseInt(response.headers.get("X-Total-Count"));
-        setTotalPages(Math.ceil(totalItems / itemsPerPage));
       } catch (error) {
         console.error("Error fetching data:", error);
       }
     };
```

Comment:
Create a new file write all of you api calls there and import those methods where ever you need.